File: 20-05-2024_Sum_of_All_Subset_Xor_Totals.java
class Solution {
    private void calculateSubsetXORSum(int[] nums, int n, int[] sum, int i, int currentXOR) {
        if (i == n) {
            sum[0] += currentXOR;
            return;
        }
        // Exclude the current element from the subset
        calculateSubsetXORSum(nums, n, sum, i + 1, currentXOR);
        // Include the current element in the subset
        calculateSubsetXORSum(nums, n, sum, i + 1, currentXOR ^ nums[i]);
    }

    public int subsetXORSum(int[] nums) {
        int[] sum = new int[1];
        calculateSubsetXORSum(nums, nums.length, sum, 0, 0);
        return sum[0];
    }
}

File: 20-05-2024_Sum_of_All_Subset_Xor_Totals.py
class Solution:
    def calculateSubsetXORSum(self, nums: List[int], n: int, i: int, currentXOR: int, sum: List[int]) -> None:
        if i == n:
            sum[0] += currentXOR
            return
        # Exclude the current element from the subset
        self.calculateSubsetXORSum(nums, n, i + 1, currentXOR, sum)
        # Include the current element in the subset
        self.calculateSubsetXORSum(nums, n, i + 1, currentXOR ^ nums[i], sum)

    def subsetXORSum(self, nums: List[int]) -> int:
        sum = [0]
        self.calculateSubsetXORSum(nums, len(nums), 0, 0, sum)
        return sum[0]
